modules/RTC/TPCUtils.js

```diff
@@ -345,44 +497,109 @@ export class TPCUtils {
      * @returns {Array<number>}
      */
-    calculateEncodingsBitrates(localVideoTrack) {
+    calculateEncodingsBitrates(localVideoTrack, codec, newHeight) {
         const videoType = localVideoTrack.getVideoType();
-        const desktopShareBitrate = this.pc.options?.videoQuality?.desktopBitrate || DESKTOP_SHARE_RATE;
-        const lowFpsScreenshare = localVideoTrack.getVideoType() === VideoType.DESKTOP
-            && this.pc._capScreenshareBitrate
-            && !browser.isWebKitBased();
-        const encodingsBitrates = this._getVideoStreamEncodings(localVideoTrack.getVideoType())
-        .map(encoding => {
-            const bitrate = lowFpsScreenshare
-                ? desktopShareBitrate
+        const desktopShareBitrate = this.pc.options?.videoQuality?.desktopbitrate || DESKTOP_SHARE_RATE;
+        const encodingsBitrates = this._getVideoStreamEncodings(localVideoTrack.getVideoType(), codec)
+        .map((encoding, idx) => {
+            let bitrate = encoding.maxBitrate;
+
+            // Single video stream.
+            if (!this.pc.isSpatialScalabilityOn() || this._isRunningInFullSvcMode(codec)) {
+                const { maxBitrate } = this._calculateActiveEncodingParams(localVideoTrack, codec, newHeight);
+
+                return idx === 0 ? maxBitrate : 0;
+            }
 
+            // Multiple video streams.
+            if (this._isScreenshareBitrateCapped(localVideoTrack)) {
+                bitrate = desktopShareBitrate;
+            } else if (videoType === VideoType.DESKTOP && browser.isChromiumBased() && !this.pc.usesUnifiedPlan()) {
                 // For high fps screenshare, 'maxBitrate' setting must be cleared on Chrome in plan-b, because
                 // if simulcast is enabled for screen and maxBitrates are set then Chrome will not send the
                 // desktop stream.
-                : videoType === VideoType.DESKTOP && browser.isChromiumBased() && !this.pc.usesUnifiedPlan()
-                    ? undefined
-                    : encoding.maxBitrate;
+                bitrate = undefined;
+            }
 
             return bitrate;
         });
 
         return encodingsBitrates;
     }
 
+    /**
+     * Returns the calculated scalability modes for the video encodings when scalability modes are supported.
+     *
+     * @param {JitsiLocalTrack} localVideoTrack The local video track.
+     * @param {CodecMimeType} codec - The codec currently in use.
+     * @param {number} maxHeight The resolution requested for the video track.
+     * @returns {Array<VideoEncoderScalabilityMode> | undefined}
+     */
+    calculateEncodingsScalabilityMode(localVideoTrack, codec, maxHeight) {
+        if (!this.pc.isSpatialScalabilityOn() || !this.codecSettings[codec].scalabilityModeEnabled) {
+            return;
+        }
+
+        // Default modes for simulcast.
+        const scalabilityModes = [
+            VideoEncoderScalabilityMode.L1T3,
```

Comment:
Is this value being the same intentional?

Reply:
Yes for simulcast, each one of the simulcast streams will have temporal scalability only and hence L1T3.